pandas/core/strings.py

```diff
@@ -2325,12 +2206,127 @@ def cat(self, others=None, sep=None, na_rep=None, join=None):
             res = Series(res, index=data.index, name=self._orig.name)
         return res
 
-    @copy(str_split)
+    _shared_docs['str_split'] = ("""
+    Split strings around given separator/delimiter.
+
+    Splits the string in the Series/Index from the %(side)s,
+    at the specified delimiter string. Equivalent to :meth:`str.%(method)s`.
+
+    Parameters
+    ----------
+    pat : str, optional
+        String or regular expression to split on.
+        If not specified, split on whitespace.
+    n : int, default -1 (all)
+        Limit number of splits in output.
+        ``None``, 0 and -1 will be interpreted as return all splits.
+    expand : bool, default False
+        Expand the splitted strings into separate columns.
+
+        * If ``True``, return DataFrame/MultiIndex expanding dimensionality.
+        * If ``False``, return Series/Index, containing lists of strings.
+
+    Returns
+    -------
+    Series, Index, DataFrame or MultiIndex
+        Type matches caller unless ``expand=True`` (see Notes).
+
+    Notes
+    -----
+    The handling of the `n` keyword depends on the number of found splits:
+
+    - If found splits > `n`,  make first `n` splits only
+    - If found splits <= `n`, make all splits
+    - If for a certain row the number of found splits < `n`,
+      append `None` for padding up to `n` if ``expand=True``
+
+    If using ``expand=True``, Series and Index callers return DataFrame and
+    MultiIndex objects, respectively.
+
+    See Also
+    --------
+     Series.str.split : Split strings around given separator/delimiter.
+     Series.str.rsplit : Splits string around given separator/delimiter, starting from 
+     the right.
+     Series.str.join : Join lists contained as elements in the Series/Index 
+     with passed delimiter. 
+     str.split : Standard library version for split.
+     str.rsplit : Standard library version for rsplit.
+
+    Examples
+    --------
+    >>> s = pd.Series(["this is a regular sentence", "this,is,comma,separated,text", np.nan])
+
+    By default, split and rsplit will return an object of the same size
+    having lists containing the split elements.
+
+    Parameter `n` can be used to limit the number of splits on the delimiter. If delimiter is
+    not specified, string is split on whitespace.
+
+    >>> s.str.split(n=2)
+    0    [this, is, a regular sentence]
+    1    [this,is,comma,separated,text]
```

Comment:
I didn't anticipate this before, but if we use comma separated values in the string, it's difficult to tell here which one has been splitted. Can be use something else like semicolon, or something like `/path/to/some/file` that looks like a real-world example and makes it obvious if it has been splitted or not?

Reply:
Yes I think the issues with commas is that it is difficult to notice the separation space, thats what i noticed - further complicated by the fact that more commas are added in where the split happens. I think the path or semi colon is a good idea - ill see which one looks best